src/Orleans/CodeGeneration/GrainInterfaceData.cs

```diff
@@ -424,7 +520,7 @@ private static bool ValidateInterfaceProperties(Type type, List<string> violatio
         {
             bool success = true;
 
-            PropertyInfo[] properties = type.GetProperties();
```

Comment:
@kumbhare Yes, you are right. My bad. This is to fail codegen in case a property is defined on the interfaces, not to check if it's async.